### DISCUSSION TITLE
Tester avec keycloak local

### DISCUSSION THREAD

**Discussion by lme-atolcd** [InseeFr/Keycloak-FranceConnect#102] [Q&A]
Comment faites-vous pour tester cette extension avec un Keycloak dispo sur localhost:8080 ?

Le problème vient de la `redirect_uri` du provider Keycloak qui ne match pas du tout avec les URL de callback du client public de FranceConnect.

* la `redirect_uri` de Keycloak : http://localhost:8080/realms/test/broker/franceconnect-particulier/endpoint
* les URL de callback de FranceConnect : https://partenaires.franceconnect.gouv.fr/fcp/fournisseur-service#sign_in

Est-ce qu'il faut obligatoirement passer par une demande d'habilitation pour avoir une URL de callback qui convienne à Keycloak ?

**Comment by clement-dufaure** [comment]
Actuellement, on utilise un client dédié pour tester FranceConnect sur lequel on a mis les url localhost format keycloak.
Du fait de l'existence de ce plugin, on pourrait peut etre essayer de négocier une url format keycloak sur la conf defaut ?

**Comment by lme-atolcd** [comment]
Il existe déjà une demande qui va dans ce sens (pour avoir une `redirect_uri` autorisée qui corresponde à une URL Keycloak).
Il s'agit de cette demande https://github.com/france-connect/service-provider-example/issues/20 et plus particulièrement [ce commentaire](https://github.com/france-connect/service-provider-example/issues/20#issuecomment-766947070).

Je vais ajouter un commentaire sur cette demande pour relancer le sujet. Sinon je tenterai ma chance avec support.usagers@franceconnect.gouv.fr.
